python/lsst/sims/catUtils/supernovae/snObject.py

```diff
@@ -454,7 +473,8 @@ def SNObjectSED(self, time, wavelen=None, bandpass=None,
 
         # Set SED to 0 beyond the model phase range, will change this if
         # SNCosmo includes a more sensible decay later.
-        if (time > self.mintime()) & (time < self.maxtime()):
+        if (time > self.mintime()) & (time < self.maxtime())\
+                & (self.modelOutSideRange=='zero'):
```

Comment:
Wait.  I think I see what you are doing, but I don't think the logic will work the way you want it.  As it is, both conditionals must be true in order for flambda to be non-zero, so, if time is between mintime and maxtime, but self.modelOutSideRange is not 'zero' (i.e. the first conditional is True but the second conditional is False), flambda will be left as an array of zeros.  You want flambda to always be non-zero if time is between mintime and maxtime, regardless of what self.modelOutSideRange is, and flambda to be zero only if both time is outside of the valid range and self.modelOutSideRange is 'zero'.  Maybe something like:

```if (time<self.mintime() || time>self.maxtim()) and self.modelOutSideRange=='zero':
    flambda = np.zeros()
else:
    actually set flambda
```